drivers/ethernet/eth_stm32_hal.c

```diff
@@ -304,38 +310,72 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 	heth = &dev_data->heth;
 
 	total_len = net_pkt_get_len(pkt);
-	if (total_len > ETH_TX_BUF_SIZE) {
+	if (total_len > (ETH_TX_BUF_SIZE * ETH_TXBUFNB)) {
```

Comment:
Move this in a dedicated commit as it impacts existing code

Reply:
Done